### ISSUE TITLE
Allow objects to be defined via DslPart

### ISSUE THREAD

**Issue by pearman** [DiUS/pact-jvm#661]
When defining large, complex, `PactDslJsonBody`'s it is difficult to reuse existing `DslPart`'s.

For example, if I have a `DslPart` that appears in multiple expected responses the simplest way (that I know of) to reuse an existing `DslPart` is something like this:
```java
private DslPart appendMyObject(DslPart dslPartIn) {
    return dslPartIn.object("myObject")
       // large object description . . .
       .closeObject();
}
```

However, I don't see why there couldn't be a method to define an object with an existing `DslPart` directly from the  `PactDslJsonBody` builder. For example:
```java
return new PactDslJsonBody()
   .object("myObject", dslForLargeObject)
   // or to match the existing naming convention in PactDslJsonArray
   .template("mySecondObject", dslForSecondLargeObject)
```

This is very similar to `template` method in `PactDslJsonArray` that allows one to define array's object type with a `DslPart`.

**Comment by uglyog** [comment]
Excellent idea!

**Comment by uglyog** [comment]
This has been released with version 3.5.18
